syntax = "proto3";

package networkSpiceMessages;

import "component.proto";

enum SimulationType {

	Frequency = 0;
	Transient = 1;

}

message Netlist {

	repeated Component component = 1;
	SimulationType type = 2;
	double begin = 3;
	double end = 4;
	uint64 points = 5;
	double step = 6;

}

message SimulationRequest {

	Netlist netlist = 1;
	repeated uint32 nodes = 2;
	
}